Add pagination section for apis (#1173)
getsentry/develop: src/docs/api/concepts.mdx

```

This is typically only needed if the endpoint is already public and we do not want to introduce a breaking change. Remember, if the endpoint is public and we remove an attribute, it is a breaking change. If you are iterating on an undocumented endpoint, return the minimal set of attributes and rely on the `expand` parameter to get more detailed information.

## Paginating responses

> >> APIs often need to provide collections of data, most commonly in the `List` standard method.
> However, collections can often be arbitrarily sized, and also often grow over time, increasing lookup time as well as the size of the responses being sent over the wire.
> Therefore, it is important that collections be paginated.

Reviewer: ```suggestion
>> APIs often need to provide collections of data, most commonly in the `List` standard method. However, collections can be arbitrarily sized, and tend to grow over time, increasing lookup time as well as the size of the responses being sent over the wire. This is why it's important for collections to be paginated.
```

> Paginating responses is a [standard practice for APIs](https://google.aip.dev/158), and we follow them at Sentry as well.
> We've seen an example of a `List` endpoint above, and these endpoints have two tell-tale signs:

Reviewer: ```suggestion
Paginating responses is a [standard practice for APIs](https://google.aip.dev/158), which Sentry follows.
We've seen an example of a `List` endpoint above; these endpoints have two tell-tale signs:
```

```json
GET /api/0/projects/{project_slug}/teams
[
  {
    "id": 1,
		"name": "Team 1",
		"slug": "team1",
  },
	{
    "id": 2,
		"name": "Team 2",
		"slug": "team2",
  }
]

```
1. The endpoint returns an array, or multiple, objects instead of just one.
2. The endpoint can sometimes end in a plural (s), but more importantly, it does __not__ end in an identifier (`*_slug`, or `*_id`).

> To paginate a response at Sentry, you can leverage the [`self.paginate`](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/base.py#L463-L476) method as part of your endpoint.

Reviewer 2: Can you add something around the line that self.paginate is the standardized way we paginate at sentry which helps us with unification of logging and monitoring?

Author: yeah sure!

Author: done!

> You can find multiple [examples of this](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/endpoints/api_applications.py#L22-L33) in the code base, and it looks something like:

Reviewer: ```suggestion
You can find multiple [examples of this](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/endpoints/api_applications.py#L22-L33) in the code base. They'll look something like:
```

```python
def get(self, request: Request) -> Response:
    queryset = ApiApplication.objects.filter(
        owner_id=request.user.id, status=ApiApplicationStatus.active
    )

    return self.paginate(
        request=request,
        queryset=queryset,
        order_by="name",
        paginator_cls=OffsetPaginator,
        on_results=lambda x: serialize(x, request.user),
    )
```

> The example above uses an offset type paginator, but you are free to use whatever paginator type suits your endpoint needs.
> There are some [existing types](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/paginator.py) that you can leverage out of the box, or you can extend the base class [`BasePaginator`](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/paginator.py#L48) and implement your own.

Reviewer: ```suggestion
The example above uses an offset type paginator, but feel free to use whatever paginator type suits your endpoint needs.
There are some [existing types](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/paginator.py) that you can leverage out of the box, or you can extend the base class [`BasePaginator`](https://github.com/getsentry/sentry/blob/24.2.0/src/sentry/api/paginator.py#L48) and implement your own.
```
